### ISSUE TITLE
`@import` seems not to be supported

### ISSUE THREAD

**Issue by blaggacao** [paulzi/svg-text-to-path#16]
Thanks for this lib.

When I import a font like this:
```html
<style>
@import url('https://fonts.googleapis.com/css2?family=Nunito:wght@1000&display=swap');
</style>
```

the browser expands it internally to this:
```html
    <style>
      /* cyrillic-ext */
      @font-face {
        font-family: "Nunito";
        font-style: normal;
        font-weight: 1000;
        font-display: swap;
        src: url(https://fonts.gstatic.com/s/nunito/v25/XRXV3I6Li01BKofIOOaBXso.woff2)
          format("woff2");
        unicode-range: U+0460-052F, U+1C80-1C88, U+20B4, U+2DE0-2DFF,
          U+A640-A69F, U+FE2E-FE2F;
      }
      /* cyrillic */
      @font-face {
        font-family: "Nunito";
        font-style: normal;
        font-weight: 1000;
        font-display: swap;
        src: url(https://fonts.gstatic.com/s/nunito/v25/XRXV3I6Li01BKofIMeaBXso.woff2)
          format("woff2");
        unicode-range: U+0301, U+0400-045F, U+0490-0491, U+04B0-04B1, U+2116;
      }
      /* vietnamese */
      @font-face {
        font-family: "Nunito";
        font-style: normal;
        font-weight: 1000;
        font-display: swap;
        src: url(https://fonts.gstatic.com/s/nunito/v25/XRXV3I6Li01BKofIOuaBXso.woff2)
          format("woff2");
        unicode-range: U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169,
          U+01A0-01A1, U+01AF-01B0, U+1EA0-1EF9, U+20AB;
      }
      /* latin-ext */
      @font-face {
        font-family: "Nunito";
        font-style: normal;
        font-weight: 1000;
        font-display: swap;
        src: url(https://fonts.gstatic.com/s/nunito/v25/XRXV3I6Li01BKofIO-aBXso.woff2)
          format("woff2");
        unicode-range: U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB,
          U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF;
      }
      /* latin */
      @font-face {
        font-family: "Nunito";
        font-style: normal;
        font-weight: 1000;
        font-display: swap;
        src: url(https://fonts.gstatic.com/s/nunito/v25/XRXV3I6Li01BKofINeaB.woff2)
          format("woff2");
        unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6,
          U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193,
          U+2212, U+2215, U+FEFF, U+FFFD;
      }
    </style>

```

but then this library fails with `Failed to read the 'cssRules' property from 'CSSStyleSheet'`

However, import rules look like this: https://developer.mozilla.org/en-US/docs/Web/API/CSSImportRule

Hence https://github.com/paulzi/svg-text-to-path/blob/146187755a7c90fd44408e7e78447b86633c70b4/providers/font-face/FontFaceSessionProvider.js#L57-L59 would need a small modification to support them.

**Comment by paulzi** [comment]
Yes, the error occurs because browsers are not allowed to read rules from other domains.
This limitation cannot be solved. The maximum that is possible is to wrap it in try-catch so that no errors occur.
If you want to use Google Fonts then use GoogleProvider (by specifying a `googleApiKey`) and disable FontFaceProvider by `useFontFace: false`

**Comment by blaggacao** [comment]
I'm not sure if I came to the same conslusion. My conslusion is that:

`rule.constructor.name === 'CSSFontFaceRule'` simply avoids looking in the right place (which would be `rule.constructor.name === 'CSSImportRule'`).

Or am I missing something?

**Comment by paulzi** [comment]
CSSImportRule has property [styleSheet](https://developer.mozilla.org/en-US/docs/Web/API/CSSImportRule/stylesheet) which is recursively walk on [this line](https://github.com/paulzi/svg-text-to-path/blob/master/providers/font-face/FontFaceSessionProvider.js#L56).
You can easily reproduce the problem in the browser console by trying to get a list of rules from an import from another domain css-file:
![изображение](https://user-images.githubusercontent.com/9402208/202870973-f3af2e34-afbb-41f2-89a5-0048248be9ae.png)

**Comment by blaggacao** [comment]
So after fixing #17, this regular `font-face` case actually works due to disk cache...

Probably one of the rare cases where a side-effect is welcome, though probably hasn't been intended this way:

![image](https://user-images.githubusercontent.com/7548295/202871412-380b62b4-49e0-404e-959b-e8ce3801b2d2.png)

Will test again the `@import` pattern and get to the bottom of the error.

**Comment by blaggacao** [comment]
When I follow the breadcrumb, I get this error:
![image](https://user-images.githubusercontent.com/7548295/202871787-efe72e7b-f5c2-43cf-b0ad-da68ac7763ad.png)

Since overall, the font face provider depends on the cache anyways (as we can see above), this issue should be fixable by fethching the relevant file (already in cache) and then reading out its rules, respectively...

Should be in the cache, already:

![image](https://user-images.githubusercontent.com/7548295/202871829-ff5d3a57-8195-4479-8d82-2e7bf9e876b9.png)

**Comment by blaggacao** [comment]
Looks like the assumption holds so far:

![image](https://user-images.githubusercontent.com/7548295/202872321-eb04c169-26e7-454b-ae3b-c260f4143103.png)

**Comment by blaggacao** [comment]
@paulzi may you help me choose my efforts: in light of the proof-of-concept PR is this going to stay on `wontfix`?

**Comment by paulzi** [comment]
I will review your request in detail tomorrow.
